0001-typedb/03-admin/04-ha.md

```diff
@@ -0,0 +1,79 @@
+---
+pageTitle: High availability
+keywords: typedb, ha, replication, failover, raft
+longTailKeywords: TypeDB replication, TypeDB high availability, high availability
+summary: TypeDB high availability guarantees and solution.
+toc: false
+---
+
+# High availability
+
+## Overview
+
+<div class="note">
+[Note]
+High availability features are available only in TypeDB Cloud.
+</div>
+
+TypeDB Cloud supports the following high availability features: 
+
+* Server clusterization,
+* [Strongly consistent model](https://en.wikipedia.org/wiki/Strong_consistency),
+* Data and schema replication with primary/secondary replicas,
+* Primary replica failover using [Raft consensus algorithm](https://en.wikipedia.org/wiki/Raft_(algorithm)),
+* Synchronous replication using [Raft consensus algorithm](https://en.wikipedia.org/wiki/Raft_(algorithm)),
+* Client level load balancing.
+
+## Replication
+
+TypeDB replication is performed at the database level. Specifically, there is a separate replication group for each 
+database. Every database has a separate replication group. There can be multiple replicas of a database, but only one 
+per server.
+
+There are two types of replicas: primary replica and secondary ones (or leader and followers in Raft terms). There can 
+be only one primary replica of a database at any given time. It is possible that primary replicas of different 
+databases can be on the same or different servers. Primary replicas location can’t be set manually, only through
+[election process](#leader-election) (as per Raft algorithm).
+
+By default, all write and read operations are performed on the primary replica. Reads from secondary replicas have 
+no guarantee of up-to-date data. But reads can be very costly (especially with rules). There is a "**read any replica**"
+transaction option to be able to read data not only from primary replica. If the option set to **True** then TypeDB 
+Client will choose a random replica to balance the load between replicas. 
+
+<div class="note">
+[Note]
+TypeDB Cloud uses ZeroMQ for replication and gRPC for replication management.
+</div>
+
+### Leader election
```

Comment:
I wonder if this is a bit too internal for the user to case? The next 3 sections are "user visible" in some way, but this one is not?

Maybe a better section that is more user visible would "Primary failover" and we can then discuss when failover happens and what that means to the user (eg. the primary may change, but clients will automatically find it when connecting for writes/strongly consistent reads)

Reply:
I think in a `failover` section you could do a small paragraph about leader election, but I think that's more of a link to a Raft resources than having to explain exactly how leader election happens.